### ISSUE TITLE
Gnome twitch crashes when typing something on games->search

### ISSUE THREAD

**Issue by fsdias** [vinszent/gnome-twitch#237]
I have gnome-twitch installed with flatpak and I am on ubuntu 16.04.

Go to games - > Search -> Type something and press enter -> crash

If I run the gnome-twitch by typing:  "/usr/bin/flatpak run --branch=master --arch=x86_64 --command=gnome-twitch com.vinszent.GnomeTwitch" crashes are less frequent. But when they happen this message appears:

[15:46:30] Critical - GLib : g_main_context_ref: assertion 'g_atomic_int_get (&context->ref_count) > 0' failed

**Comment by vinszent** [comment]
I was encountering this problem as well and I think I've fixed it now. Can you confirm?

**Comment by fsdias** [comment]
Sorry, I just updated the app with flatpak but I'm still encountering the same issue

**Comment by vinszent** [comment]
What version are you running? I'm uncertain about how often Flatpak builds off master.

**Comment by fsdias** [comment]
In the about section this is the reported version: r987.f6862e0

**Comment by vinszent** [comment]
Does this happen when searching channels?

**Comment by fsdias** [comment]
I searched for channels for ~ 5 minutes and it didn't crash. In the same session I switched to searching for games and it crashed after two searches.

**Comment by vinszent** [comment]
I'm unable to replicate the error you're encountering and it's wierd because searching for channels and searching for games use virtually the same code.

It would be really helpful if I could get a stacktrace from you. You'll need to run the following commands:

```sh
$ flatpak install com.vinszent.GnomeTwitch com.vinszent.GnomeTwitch.Debug
$ flatpak run -d --command=sh com.vinszent.GnomeTwitch
$ gdb gnome-twitch
(gdb) run
(gdb) backtrace (When it crashes)
```
And then paste the stacktrace here or on a pastebin somewhere if it's long

**Comment by fsdias** [comment]
flatpak install com.vinszent.GnomeTwitch com.vinszent.GnomeTwitch.Debug

returns:

error: Remote "com.vinszent.GnomeTwitch" not found

Typo somewhere?

**Comment by vinszent** [comment]
Oh right yes, it should be `com.vinszent.GnomeTwitch-origin` i.e. `flatpak install com.vinszent.GnomeTwitch-origin com.vinszent.GnomeTwitch.Debug`

**Comment by vinszent** [comment]
Sorry, I'm not super familiar with Flatpak but from a fresh install this is what I ran:

```sh
flatpak install --from https://dl.tingping.se/flatpak/gnome-twitch.flatpakref
flatpak install gnome org.gnome.Sdk//3.22
flatpak install com.vinszent.GnomeTwitch-origin com.vinszent.GnomeTwitch.Debug
flatpak run -d --command=sh com.vinszent.GnomeTwitch
```

**Comment by fsdias** [comment]
After running:

flatpak run -d --command=sh com.vinszent.GnomeTwitch

this appears:

sh-4.3$ 

should I type anything else or just run gnome twitch?

**Comment by vinszent** [comment]
Yeah now you can run:
```sh
gdb gnome-twitch
(gdb) run
(gdb) backtrace (When it crashes
```

**Comment by fsdias** [comment]
Gnome twitch froze but did not crash as before:

(gdb) run
Starting program: /app/bin/gnome-twitch 
warning: Error disabling address space randomization: Operation not permitted
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/libthread_db.so.1".
[New Thread 0x7f23f851b700 (LWP 9)]
Gtk-Message: Failed to load module "unity-gtk-module"
[18:38:33] Message - GNOME-Twitch : {GtApp:380} Startup
[18:38:33] Message - GNOME-Twitch : {GtApp:349} Activate
[New Thread 0x7f23e7fff700 (LWP 16)]
[New Thread 0x7f23e7548700 (LWP 17)]
[18:38:33] Message - GNOME-Twitch : {GtPlayer:132} Loading chat settings
[18:38:33] Message - GNOME-Twitch : {GtPlayer:1061} Loaded player backend 'GStreamer Cairo player backend'
[18:38:33] Message - GNOME-Twitch : {GtPlayerBackendGstreamerCairo:246} Init
[18:38:33] Warning - GNOME-Twitch : cannot set NULL uri
[18:38:34] Message - GNOME-Twitch : {GtApp:147} Succesfully fetched oauth info with name 'filipesdias', id '118352132' and oauth token 'zbo90xyjm691m3d56z9085q5y69c1u'
[18:38:34] Message - GNOME-Twitch : {GtApp:182} Succesfully fetched user info with name 'filipesdias', id '118352132' and oauth token 'zbo90xyjm691m3d56z9085q5y69c1u'
[18:38:35] Message - GNOME-Twitch : {GtFollowsManager:519} Loaded '5' follows from Twitch
[Thread 0x7f23e7fff700 (LWP 16) exited]
[Thread 0x7f23e7548700 (LWP 17) exited]
**
ERROR:../src/gt-search-game-container.c:117:fetch_items: assertion failed: (!utils_str_empty(priv->query))
[New Thread 0x7f23e7548700 (LWP 21)]
[New Thread 0x7f23e50da700 (LWP 22)]
[New Thread 0x7f23f5f66700 (LWP 12)]
[New Thread 0x7f23f6767700 (LWP 11)]
[New Thread 0x7f23f6f68700 (LWP 10)]

Program received signal SIGABRT, Aborted.
[Switching to Thread 0x7f23e7548700 (LWP 21)]
0x00007f2408474598 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:55
55	  return INLINE_SYSCALL (tgkill, 3, pid, selftid, sig);
(gdb)

**Comment by vinszent** [comment]
Ah ok this is another bug, but it's not related to your original bug. I've pushed a quick fix that corrects it. Maybe in about 30 minutes or so once the Flatpak package has been built, if you could update and then run the `gdb` again with the same instructions as before, that would be great. You should have version `r988.22901ab`.

**Comment by fsdias** [comment]
Ok, I will report back as it's available

**Comment by dengelt** [comment]
I built the current master (not flatpak) and could also not reproduce the bug, but what I did find is that after I search for a game and click on it, the search bar becomes greyed-out and I can't modify the search anymore until I restart gnome-twitch.

**Comment by vinszent** [comment]
@dengelt 
This is intentional, you can press the back button and edit the search again. My idea behind this is that you have a persistent search but can't change it when you are in a game (as this would be hard to handle). In older versions of GT the search bar would close and you would have to retype your entire search if you wanted to change it.

**Comment by dengelt** [comment]
Ok, I didn't notice the back button in the top left. But this doesn't work if I close the search bar by pressing the search icon before pressing the back button. If I reset the search this way, the search field will be blank and no back button will be available, but the search bar will still be greyed out.

**Comment by vinszent** [comment]
Good find! I'll get on it and fix it.

**Comment by fsdias** [comment]
The lastest version just came out with flatpak. I browsed around for about five minutes with no crashes. But eventually gnome twitch froze:

[18:41:20] Message - GNOME-Twitch : {GtIrc:997} Joining with channel='#doom49'
[New Thread 0x7f82b8a9c700 (LWP 57)]
[New Thread 0x7f82aa7fe700 (LWP 56)]
[New Thread 0x7f82baf0a700 (LWP 54)]
[New Thread 0x7f82aafff700 (LWP 53)]
[New Thread 0x7f829bfff700 (LWP 49)]
[New Thread 0x7f82a8809700 (LWP 40)]
[New Thread 0x7f82a980a700 (LWP 39)]
[New Thread 0x7f82bbfff700 (LWP 31)]
[New Thread 0x7f82c9966700 (LWP 12)]
[New Thread 0x7f82ca167700 (LWP 11)]
[New Thread 0x7f82ca968700 (LWP 10)]

Program received signal SIGSEGV, Segmentation fault.
0x0000000100000001 in ?? ()

**Comment by fsdias** [comment]
I remember that the crashes tended to occur when I was trying to click on the search bar. I wonder if the problem I was experiencing was somehow related to the one you just fixed. Anyway, aside from the issue above I'm no longer experiencing crashes.

**Comment by vinszent** [comment]
Could you type `backtrace` next time this happens so the entire stack where it crashed is printed?

**Comment by fsdias** [comment]
gnome-twitch no longer crashes. When should I type backtrace...? This is what I am running:

flatpak run -d --command=sh com.vinszent.GnomeTwitch
gdb gnome-twitch
run

**Comment by vinszent** [comment]
Sure, after you type `run` it will run and then when it crashes you can type the terminal, i.e. in gdb, `backtrace` and it should print a stacktrace of where it crashed.

Also I just saw your second comment. I think the search bar thing is just a coincidence. There are a few mutli threading issues that I think are causing these random crashes that I need to iron out (seems like one of the libraries might not be as thread safe as it claims to be 😝 ). Anyways hopefully the amount of crashes is less than before 🤞 .

**Comment by fsdias** [comment]
Ah, I hadn't realized I could still type. Here it is:

#0  0x0000000000000000 in ?? ()
#1  0x00007f57e96052be in ?? () from /lib/libgtk-3.so.0
#2  0x00007f57e9605503 in ?? () from /lib/libgtk-3.so.0
#3  0x00007f57e961b73a in ?? () from /lib/libgtk-3.so.0
#4  0x00007f57e961a7ac in ?? () from /lib/libgtk-3.so.0
#5  0x00007f57e961b565 in ?? () from /lib/libgtk-3.so.0
#6  0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#7  0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#8  0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#9  0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#10 0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#11 0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#12 0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#13 0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#14 0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#15 0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#16 0x00007f57e961b53a in ?? () from /lib/libgtk-3.so.0
#17 0x00007f57e96029f6 in ?? () from /lib/libgtk-3.so.0
#18 0x00007f57e89bc575 in g_closure_invoke () from /lib/libgobject-2.0.so.0
#19 0x00007f57e89cdd81 in ?? () from /lib/libgobject-2.0.so.0
#20 0x00007f57e89d69af in g_signal_emit_valist () from /lib/libgobject-2.0.so.0
#21 0x00007f57e89d6cd2 in g_signal_emit () from /lib/libgobject-2.0.so.0
#22 0x00007f57e91f5e27 in ?? () from /lib/libgdk-3.so.0

**Comment by vinszent** [comment]
Right, these kind of crashes irk me. They are internal crashes in GTK and are pretty much out of my control.  I get them occassionaly as well but really they shouldn't occur all that often...

**Comment by fsdias** [comment]
Let me know if there is anything else I can help you with

**Comment by vinszent** [comment]
Awesome, thanks for your help so far! I'm really trying to make GT as stable and error prone as possible so all the information you posted is a great help. Again if you encounter any crashes in the future it helps a lot to get a stacktrace.

I'll leave the issue open for a week or something. If it starts crashing again just leave a comment here or re-open the issue in the future 😄 .

**Comment by dengelt** [comment]
> I'm really trying to make GT as stable and error prone as possible

FYI, "error prone" means that errors are likely to occur ;)

**Comment by fsdias** [comment]
Happy to help

**Comment by vinszent** [comment]
@dengelt Thanks, fixed it now.

**Comment by Techwolf** [comment]
While I am late to this party, when I try to run under gdb, it takes about three to four minutes for the first window to popup. And about 6GB of memory is used. Thankfully, after getting to the first window, it runs fine with no preformence problems. I am just saying this so others don't think gdb gnome-twitch is broken like I did at first.

**Comment by vinszent** [comment]
@Techwolf Hmm that does sound quite wierd. Personally I've never had any problems when using gdb with GT, either memory usage or startup times. Can you provide me with some details about your system? I.e. distro, are you using any GT packages, GT version, etc.

**Comment by dengelt** [comment]
@Techwolf That doesn't seem normal. Are there any strange messages? Does the memory gradually fill up or does it just suddenly consume that much memory? How did you get gnome-twitch? Does this also happen with other applications? Try `gdb gedit` for example.

**Comment by Techwolf** [comment]
@vinszent @dengelt I did try looking for a forum or other support area. Glitter is 404 and nothing else was found. So I will sorta hijack this issue.

I did figure out a cause of my gdb problem, it was looking at this project home page and notice a depends that I had sorta forgotten about. Webkit. That package alone takes about an hour and half to compile on my AMD 8 core CPU. I am quiet sure is that webkit that is taken all the time and memory. I run gentoo with splitdebug enabled globalwide. Gentoo is a good distro for developing on and using at the same time.

One reason I was glad to get this desktop version of twitch.tv going is twitch.tv built in player can really suck balls at times. While I can manually run mpv via streamlink, I wanted something more then that. And NOT another <censored> embedded browser app pretending to be a desktop program. </rant>

While this program still uses an embedded browser, it only used for browsing. I love the fact it is NOT used for viewing. Using mpv and a native chat is nearly perfect. :-)
